lib/graphiti/sideload.rb

```diff
@@ -177,7 +177,8 @@ def association_name
     end
 
     def resource_class
-      @resource_class ||= infer_resource_class
+      @cons_resource_class ||= (@resource_class.is_a?(String) ? @resource_class.constantize : @resource_class) ||
```

Comment:
@jkeen I'm not sure about the `@cons_resource_class` variable name but `@resource_class` directly assigns in line 52.  Probably the best variant would be to rename `@resource_class` to something like `@resource_class_name` and then use `@resource_class` here for memorization. What do you think?

Reply:
That is how ActiveRecord does it for its relations. You set the relation `class_name` property as a string. It's done to omit a situation when one of the classes is not defined yet. Please pay attention to the test. When Ruby interprets `UserResource` the `BookResource` is not defined yet that's why if you try to use the class directly it causes an `uninitialized constant` error.
Usually, developers do not face this problem because `infer_resource_class` actually does the "lazy constantize" based on the relation name. But it can be a problem when you use custom relation names (like was done in the test) which is pretty common when building relations between models in different namespaces.

Reply:
@bguban I updated those instance variable names and the tests now pass

Reply:
@jkeen well done! Thank you! LGTM